pkg/buildpipeline/pipeline.go

```diff
@@ -155,6 +155,61 @@ func (s *Stage) TaskName() string {
 	return strings.ToLower(strings.NewReplacer(" ", "-").Replace(s.Name))
 }
 
+// Task names need to be RFC 1035/1123 compliant DNS labels, so we mangle the
+// input to make it compliant. Results should match the following regex and be
+// no more than 63 characters long:
+//     [a-z]([-a-z0-9]*[a-z0-9])?
+// cf. https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+// Suffix is assumed to be alphanumeric and non-empty.
+func createTaskName(pipelineIdentifier string, stageName string, buildIdentifier string, suffix string) string {
+	const MAX_LABEL_LENGTH = 63
+	MAX_BODY_LENGTH := MAX_LABEL_LENGTH - len(suffix) - 1 // Add an extra hyphen before the suffix
+
+	// Having `stage` and `build` literally in the output means the body is
+	// guaranteed to be non-empty. Would it be better to fail if the output is
+	// `stage-build-$suffix` because everything else is invalid?
+	body := fmt.Sprintf("%s-stage-%s-build-%s", pipelineIdentifier, stageName, buildIdentifier)
```

Comment:
For example, see my test case using `. --- .` as a stage name of a case where failing might make more sense.

Reply:
Depending on how long our suffixes end up being we could use something like `%s-s-%s-b-%s` to reclaim a few characters.

Also, it feels a little clearer to me to use a pipeline-build-stage order, so that any sorted output would automatically put tasks related to a single build together vs putting stages together across multiple builds.

Reply:
and I'm utterly indifferent to the order - I was just throwing stuff around. =)